src/caja-desktop-window.c

```diff
@@ -230,15 +303,21 @@ unrealize (GtkWidget *widget)
     CajaDesktopWindow *window;
     CajaDesktopWindowPrivate *details;
     GdkWindow *root_window;
+    GdkDisplay *display;
 
     window = CAJA_DESKTOP_WINDOW (widget);
     details = window->details;
+    display = gdk_screen_get_display (gdk_screen_get_default());
 
-    root_window = gdk_screen_get_root_window (
+    /*Avoid root window on wayland-it's not supposed to work*/
+    if (GDK_IS_X11_DISPLAY (display))
+    {
+        root_window = gdk_screen_get_root_window (
                       gtk_window_get_screen (GTK_WINDOW (window)));
 
-    gdk_property_delete (root_window,
+        gdk_property_delete (root_window,
                          gdk_atom_intern ("CAJA_DESKTOP_WINDOW_ID", TRUE));
```

Comment:
should align with the opening parenthesis from the line above

Reply:
In most of the existing code, first letter of second line of something like this is under the first letter after the (  and I've been advised elsewhere to align it that way, so doing that here too